### ISSUE TITLE
A question about FBA and FVA

### ISSUE THREAD

**Issue by Erique29** [jotech/gapseq#79]
Hi ,
  After constructing metabolic models using this wonderful tool, I want to validate the accuracy of my models using BIOLOG plates, like the paper 'The functional repertoire contained within the native microbiota of the model nematode _Caenorhabditis elegans_'. I used the data 'MYB11 and MYB71' of this published study to test. I referred to your method in [Lactic acid bacteria in yogurt production](https://gapseq.readthedocs.io/en/latest/tutorials/yogurt.html), but have not find the right result of carbon sources and by-products in Fig. S9. Could you help me with this? THANKS!

**Comment by jotech** [comment]
Hi @Erique29,
great idea to validate the models using phenotypic data! If you already have the biolog data available, I would recommend to use a check for electron donor instead of standard FBA with biomass optimization. It's more close to the actual biolog experiment to test the capacity of a compound to serve as electron source for electron carriers like NAD or quinones. Some example code for this can be found in the [code repository](https://github.com/Waschina/gapseqEval/blob/main/CarbonSourceTest/CarbonSourceTest.R#L67:L121) of the gapseq paper. Do you think this could be of any help in your case?
In the past, I found it not always easy to discriminate between growth/non-growth using biolog, a foldchange > 1 of OD corrected values (early vs. late) worked best if I remember correctly. What is your approach here?
Thanks for your feedback!

**Comment by Erique29** [comment]
Thanks @jotech !
  This is indeed a good way to validate the BIOLOG data using the electron transfer which works the same as BIOLOG. Unfortunately, I do not have BIOLOG data now. I will start the BIOLOG experiment in next few weeks. And I will communicate with you about the phenomenon you have found.
  By the way, I would like to validate the models using both the electron transfer and the FBA to have a mutual confirmation. And I'm also interested in the carbon sources and metabolic byproducts. How can I modify the method I used above to get the similar results as Fig. S9? It actually troubled me for many days.

**Comment by jotech** [comment]
Hi @Erique29, the potential carbon sources and byproducts from Figure S9 were determined by flux variability analysis (FVA). If there was a  non-zero flux of the respective exchange reaction while guaranteeing at least 10% of the maximum growth yield, then a compound was assumed to be a carbon source (flux<0) or byproduct (flux>0). Does it makes sense to you?

**Comment by Erique29** [comment]
Hi, it does make sense to me. I will try it! One more question, I tried MEMOTE to score my model, but I found the annotation-genes were all zero. And the total score was 77. Is there any problems in my model? Thanks!
[index.zip](https://github.com/jotech/gapseq/files/6681882/index.zip)

**Comment by Waschina** [comment]
Hi!
No, this is nothing to worry about. The reason is that gapseq follows a unique approach to gene-identfiers in model. The gene names in gapseq indicate the contig-ID and the position of genes in the input genome sequence (see for instance also at the discussion in the issue #71).

**Comment by Erique29** [comment]
Yeah, It explains the matter. Thanks for your help!

**Comment by jotech** [comment]
please reopen if there are still things unclear :)

**Comment by Erique29** [comment]
Hi @jotech,
I finished the BIOGLOG experiment this week, and I found that the foldchange > 1 of OD590  of some wells were not easy to discriminate. Because they were lavender like some wells with foldchange < 1. According to my observation, foldchange >2 of OD590 is better to discriminate the growth. Because the color of the wells (foldchange >2) were all dark purple after 48 h.
At last, I used BIOLOG GEN III, and my initial OD590 of bacterial suspension was 0.05.
